### PR TITLE
Node for publishing all visible markers

### PR DESCRIPTION
**Pull request by jdlangs**
Is the plan to make this the default node that people are directed to and then add the filtering functionality? Maybe there can be an Indigo/Jade official release with this behavior?

### CODE REVIEW

**Comment by bmagyar** [comment]
I'll test it tonight and if all are well, I'll merge it right away. 
The answers to your questions are in order are as follows: yes, yes :)
I am planning to make this the main executable, maybe even to eliminate the old ones.

File: aruco_ros/launch/marker_pub.launch [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<launch>
+
+    <arg name="markerSize"      default="0.05"/>    <!-- in m -->
+    <arg name="side"             default="left"/>
+    <arg name="ref_frame"       default="base"/>  <!-- leave empty and the pose will be published wrt param parent_name -->
+
+
+    <node pkg="aruco_ros" type="marker_pub" name="aruco_marker_publisher">
```
Comment (bmagyar) [inline aruco_ros/launch/marker_pub.launch]: The type here should be the name of the binary that you compile the node into: `marker_publisher`
The launch file as of now fails

**Comment by bmagyar** [comment]
Apart from the launch file typo I mentioned above, all seems fine. 
Once this is merged I will continue working on it during next week in order to reach a point where it is releasable.

Thanks for the contribution!

**Comment by phobon** [comment]
Yep, sorry about that. I did some final renaming but forgot about the launch file. It should be all set now.

**Comment by bmagyar** [comment]
Merging... :+1:

**Comment by nbaddorf** [comment]
Hey, thanks so much for you work on this! I have spent the past couple of days trying to get this system to work, and I finally achieved it, but I have, probably a simple issue, where the `/aruco_marker_publisher/markers`  topic is outputting: `header:
  seq: 695
  stamp: 
    secs: 1614385501
    nsecs:  30457688
  frame_id: "eye"
markers: 
  - 
    header: 
      seq: 0
      stamp: 
        secs: 1614385501
        nsecs:  30457688
      frame_id: "eye"
    id: 582
    pose: 
      pose: 
        position: 
          x: -0.115686513484
          y: 0.0845252200961
          z: 0.507478535175
        orientation: 
          x: 0.970179223221
          y: -0.066177375452
          z: 0.118988815579
          w: 0.200535512005
      covariance: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
    confidence: 1.0
---
`
The problem is that there is a "-" between the header and pose. causing me not to be able to read the pose data in a program. Could you help me fix this?

**Comment by nbaddorf** [comment]
Sorry, about the bold text in the previous msg, the dash that is causing my problems, is making the github code tag not work, and I cant seem to fix it. But for clarification there is a "-" in the output between the "markers:" and the "header". Thanks!

**Comment by v-lopez** [comment]
How are you subscribing to the topic? C++ or Python subscriber?

**Comment by nbaddorf** [comment]
Hi, I was trying to subscribe with python. In the end I was able to figure out that if I made a new variable with the condense of msg.markers[0], it would let me use that new var to read: newMsg.id and newMsg.pose. Thanks for making this system!

**Pull request merged**